weave/weave_client.py

```diff
@@ -818,6 +818,19 @@ def op_fn(*args, **kwargs):  # type: ignore
 
 
 def redact_sensitive_keys(obj: typing.Any) -> typing.Any:
+    # We should NEVER mutate reffed objects.
+    #
+    # 1. This code builds new objects that no longer have refs
+    # 2. Even if we did an in-place edit, that would invalidate the ref (since
+    # the ref is to the object's digest)
+
+    try:
```

Comment:
Is there a reason we need this broad try/except?

Reply:
I was under the impression this was risky, but you are correct, it should be safe. We actually recursively do this check immediately after this call so we should be good. Remove the try/except